### DISCUSSION TITLE
Example Script Sensors

### DISCUSSION THREAD

**Discussion by joshuar** [joshuar/go-hass-agent#192] [Show and tell]
Hey folks,

Use this topic to share your scripts (or links to them) you are using in Go Hass Agent for the [script sensor](https://github.com/joshuar/go-hass-agent?tab=readme-ov-file#script-sensors) feature!

**Comment by AdamVig** [comment]
**Webcam in use**

The built-in PipeWire-based webcam usage detection apparently only works for certain apps, so I built my own solution that polls `/dev/video*` with `lsof` to check for usage, ignoring persistent usage by processes like PipeWire and WirePlumber.

https://github.com/AdamVig/dotfiles/blob/b0c1d42c26f6c1f5839ef0e7a34bb1e1ec1d9fb0/config/go-hass-agent/webcam-in-use

Being a polling solution, it's obviously slower than a solution based on PipeWire or DBus events would be, but it has the advantage of working across all desktop environments.

**Comment by AdamVig** [comment]
**Audio output**

Creates a sensor containing the "nick" (human-friendly name) of the current default sink according to WirePlumber (via `wpctl`).

https://github.com/AdamVig/dotfiles/blob/b0c1d42c26f6c1f5839ef0e7a34bb1e1ec1d9fb0/config/go-hass-agent/audio-output

I use this for an "Audio output" card on my desk dashboard.
